rdrf/rdrf/admin.py

```diff
@@ -178,6 +183,87 @@ def get_readonly_fields(self, request, obj=None):
         return () if obj is None else ("code",)
 
 
+class ActivationKeyExpirationListFilter(admin.SimpleListFilter):
+    title = _('activation key expired')
+    parameter_name = 'activation_key_expired'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('True', _('Yes')),
+            ('False', _('No')),
+        ]
+
+    def queryset(self, request, queryset):
+        expired_profiles = [profile.id for profile in queryset if profile.activation_key_expired()]
+
+        if self.value() == 'True':
+            return queryset.filter(id__in=expired_profiles)
+        elif self.value() == 'False':
+            return queryset.exclude(id__in=expired_profiles)
+        else:
+            return queryset
+
+
+def resend_activation_mail(profile, site, request=None):
+    if not hasattr(profile.user, 'registrationprofile') or profile.activated:
+        return False
+
+    profile.create_new_activation_key()
+    profile.send_activation_email(site, request)
+
+    return True
+
+
+class CustomRegistrationProfileAdmin(RegistrationAdmin):
+    list_display = ('user', 'custom_activation_key_expired', 'activated')
+    list_filter = ['activated', ActivationKeyExpirationListFilter]
+
+    @admin.display(description="Activation key expired")
+    def custom_activation_key_expired(self, obj):
+        max_expiry_days = datetime.timedelta(days=settings.ACCOUNT_ACTIVATION_DAYS)
+        activation_date = obj.user.date_activated
+        expiration_date = (obj.user.date_joined if activation_date is None else activation_date) + max_expiry_days
+
+        return obj.activated or expiration_date <= datetime.datetime.now()
+
+    def activate_user(self, activation_key):
+        sha256_re = re.compile('^[a-f0-9]{40,64}$')
+
+        def activate(user_profile):
+            user = user_profile.user
+            user.is_active = True
+            user_profile.activated = True
+
+            with transaction.atomic():
+                user.save()
+                user_profile.save()
+
+            return user
+
+        if sha256_re.search(activation_key):
+            profile = RegistrationProfile.objects.filter(activation_key=activation_key)
+            if not profile.exists():
+                return False, False
+
+            profile = profile.first()
+            if not self.custom_activation_key_expired(profile):
+                return activate(profile), True
+
+        return False, False
+
+    def activate_users(self, request, queryset):
+        for profile in queryset:
+            self.activate_user(profile.activation_key)
+
+    def resend_activation_email(self, request, queryset):
+        site = get_current_site(request)
+        for profile in queryset:
+            resend_activation_mail(profile, site, request)
```

Comment:
A couple of things with this `resend_activation_mail` function:
* It's only used once (here), so is there any benefit to defining it at the global scope rather than just inlining the code here?
* If you feel there is a benefit to defining it separately, can we rename the methods - I know they are slightly different (email vs mail), but can be easily be mistaken for the same name and result in confusion. Otherwise if we inline it as described above, then we don't have to think of another name :)
* The global function `resend_activation_mail` also returns True/False which isn't checked here